hack/sshtool/azure.go

```diff
@@ -0,0 +1,169 @@
+	for i := 0; i < 3; i++ {
+		nicName := fmt.Sprintf("%s-master%d-nic", s.infraID, i)
+
+		nic, err := s.interfaces.Get(ctx, s.clusterResourceGroup, nicName, "")
+		if err != nil {
+			return err
+		}
+
+		disableNIC(&nic)
+
+		s.log.Printf("updating %s", nicName)
+		err = s.interfaces.CreateOrUpdateAndWait(ctx, s.clusterResourceGroup, nicName, nic)
+		if err != nil {
+			return err
+		}
+	}
+
+	lbName := s.infraID + "-internal-lb"
+
+	lb, err := s.loadBalancers.Get(ctx, s.clusterResourceGroup, lbName, "")
+	if err != nil {
+		return err
+	}
+
+	disableLB(&lb)
+
+	s.log.Printf("updating %s", lbName)
+	return s.loadBalancers.CreateOrUpdateAndWait(ctx, s.clusterResourceGroup, lbName, lb)
+}
+
+// enable adds load balancer and master nic configuration associated with SSH
+func (s *sshTool) enable(ctx context.Context) error {
+	lbName := s.infraID + "-internal-lb"
+
+	lb, err := s.loadBalancers.Get(ctx, s.clusterResourceGroup, lbName, "")
+	if err != nil {
+		return err
+	}
+
+	enableLB(&lb)
+
+	s.log.Printf("updating %s", lbName)
+	err = s.loadBalancers.CreateOrUpdateAndWait(ctx, s.clusterResourceGroup, lbName, lb)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < 3; i++ {
+		nicName := fmt.Sprintf("%s-master%d-nic", s.infraID, i)
+
+		nic, err := s.interfaces.Get(ctx, s.clusterResourceGroup, nicName, "")
+		if err != nil {
+			return err
+		}
+
+		enableNIC(&nic, &lb, i)
+
+		s.log.Printf("updating %s", nicName)
+		err = s.interfaces.CreateOrUpdateAndWait(ctx, s.clusterResourceGroup, nicName, nic)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func disableNIC(nic *mgmtnetwork.Interface) {
+	backendAddressPools := make([]mgmtnetwork.BackendAddressPool, 0, len(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools))
+	for _, p := range *(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools {
+		if !strings.HasPrefix(*p.Name, "ssh-") {
+			backendAddressPools = append(backendAddressPools, p)
+		}
+	}
+
+	(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = &backendAddressPools
+}
+
+func enableNIC(nic *mgmtnetwork.Interface, lb *mgmtnetwork.LoadBalancer, i int) {
+	disableNIC(nic)
+
+	*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = append(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools, mgmtnetwork.BackendAddressPool{
+		ID: to.StringPtr(fmt.Sprintf("%s/probes/backendAddressPools/ssh-%d", *lb.ID, i)),
+	})
+}
+
+func disableLB(lb *mgmtnetwork.LoadBalancer) {
+	backendAddressPools := make([]mgmtnetwork.BackendAddressPool, 0, len(*lb.BackendAddressPools))
+	for _, p := range *lb.BackendAddressPools {
+		if !strings.HasPrefix(*p.Name, "ssh-") {
+			backendAddressPools = append(backendAddressPools, p)
+		}
+	}
+	*lb.BackendAddressPools = backendAddressPools
+
+	loadBalancingRules := make([]mgmtnetwork.LoadBalancingRule, 0, len(*lb.LoadBalancingRules))
+	for _, c := range *lb.LoadBalancingRules {
+		if !strings.HasPrefix(*c.Name, "ssh-") {
```

Comment:
not sure why i got nil names..
```suggestion
		if p.Name != nil &&!strings.HasPrefix(*c.Name, "ssh-") {
```